src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -52,9 +52,11 @@ class BookStep : public StepImp<TIn, TOut, BookStep<TIn, TOut>>
     Book book_;
     AccountID strandSrc_;
     AccountID strandDst_;
-    // Charge transfer fees whan the prev step redeems
+    // Charge transfer fees when the prev step redeems
     Step const* const prevStep_ = nullptr;
     bool const ownerPaysTransferFee_;
+    bool const removeSelfCrossingOffers_;
+    Quality const limitQuality_;
```

Comment:
I'd rename `limitQuality_` -> `limitSelfCrossingQuality_` and make it a boost::optional. I'd also get rid of the `removeSelfCorssingOffers_` bool member and use the optional to decide if offers should be removed.